File: android/src/main/kotlin/video/api/flutter/livestream/utils/DeviceOrientationExtensions.kt
package video.api.flutter.livestream.utils

import io.flutter.embedding.engine.systemchannels.PlatformChannel

fun PlatformChannel.DeviceOrientation.toSerializedString(): String {
    return when (this) {
        PlatformChannel.DeviceOrientation.PORTRAIT_UP -> "portraitUp"
        PlatformChannel.DeviceOrientation.PORTRAIT_DOWN -> "portraitDown"
        PlatformChannel.DeviceOrientation.LANDSCAPE_LEFT -> "landscapeLeft"
        PlatformChannel.DeviceOrientation.LANDSCAPE_RIGHT -> "landscapeRight"
    }
}

fun String.toDeviceOrientation(): PlatformChannel.DeviceOrientation {
    return when (this) {
        "portraitUp" -> PlatformChannel.DeviceOrientation.PORTRAIT_UP
        "portraitDown" -> PlatformChannel.DeviceOrientation.PORTRAIT_DOWN
        "landscapeLeft" -> PlatformChannel.DeviceOrientation.LANDSCAPE_LEFT
        "landscapeRight" -> PlatformChannel.DeviceOrientation.LANDSCAPE_RIGHT
        else -> throw UnsupportedOperationException("Could not deserialize device orientation: $this")
    }
}

File: lib/src/extensions/device_orientation_extensions.dart
import 'package:flutter/services.dart';
import 'package:native_device_orientation/native_device_orientation.dart';

extension NativeDeviceOrientationInDegress on NativeDeviceOrientation {
  int toDegress() {
    return deviceOrientation!.toDegress();
  }
}

extension DeviceOrientationInDegress on DeviceOrientation {
  int toDegress() {
    final Map<DeviceOrientation, int> degreesForDeviceOrientation =
        <DeviceOrientation, int>{
      DeviceOrientation.portraitUp: 0,
      DeviceOrientation.landscapeRight: 90,
      DeviceOrientation.portraitDown: 180,
      DeviceOrientation.landscapeLeft: 270,
    };
    return degreesForDeviceOrientation[this]!;
  }
}
